addons/website_sale/static/src/js/website_sale.js

```diff
@@ -216,6 +217,44 @@ publicWidget.registry.WebsiteSale = publicWidget.Widget.extend(VariantMixin, {
         return VariantMixin.getSelectedVariantValues.apply(this, arguments);
     },
 
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * @public
+     * @param $el {Element}
+     */
+    resizeRibbons: function ($el) {
+        var self = this;
+        if ($el) {
+            self.$ribbonWrapper = $($el);
+            _resizeRibbon();
+        } else {
+            _.each($('div.ribbon-wrapper:visible'), function ($ribbonWrapper) {
+                self.$ribbonWrapper = $($ribbonWrapper);
+                _resizeRibbon();
+            });
+        }
+
+        function _resizeRibbon() {
+            var $ribbon = self.$ribbonWrapper.find('a.ribbon');
+            var ribbonLength = $ribbon.text().length;
+            if (ribbonLength >= 10) {
+                var size = '145px';
+                self.$ribbonWrapper.css({
+                    'width': size,
+                    'height': size,
+                });
+                $ribbon.css({
+                    'top': '25px',
+                    'width': size,
+                    'font-size': '85%',
+                });
+            }
+        }
+    },
```

Comment:
This is very crappy :hankey:  but I don't have better idea for this part of the specs: 

> "we need a smart ribbon. If the word is longer, move down the ribbon to have everything visible." 

I need your expertise on that. Do you have a better idea ? Maybe only with CSS ?